### PR TITLE
feat(tools): add PDM to tools

### PR DESCRIPTION
**Pull request by secustor**
- Adds PDM for usage by the new PEP621 manager
- replaces h3py with h5py for tests as the former is no longer available

### CODE REVIEW

File: src/usr/local/buildpack/tools/v2/pdm.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# shellcheck source=/dev/null
+. "$(get_buildpack_path)/utils/python.sh"
+
+function link_tool () {
+  post_install
+  [[ -n $SKIP_VERSION ]] || poetry --version
```
Comment (rarkins) [inline src/usr/local/buildpack/tools/v2/pdm.sh:8]:
```suggestion
  [[ -n $SKIP_VERSION ]] || pdm --version
```

??

Comment (secustor) [reply src/usr/local/buildpack/tools/v2/pdm.sh:8]: Copy and paste error

File: src/usr/local/buildpack/tools/v2/pdm.sh
Comment (rarkins) [inline src/usr/local/buildpack/tools/v2/pdm.sh:1]: How/where is pdm actually installed here? I only see python

Comment (secustor) [reply src/usr/local/buildpack/tools/v2/pdm.sh:1]: Maybe I'm understanding the functionality of base not correctly but my expectation has been that adding this file is enough to install it via pip. 🤔

Comment (rarkins) [reply src/usr/local/buildpack/tools/v2/pdm.sh:1]: I've got to admit it looks that way in https://github.com/containerbase/base/blob/main/src/usr/local/buildpack/tools/v2/pipenv.sh for example. @viceice @Chumper is there some magic glue based on file name?

File: test/Dockerfile.jammy [hunk 2]
```diff
@@ -137,6 +137,8 @@ RUN install-tool pipenv 2023.4.29
 RUN install-tool hashin 0.17.0
 # renovate: datasource=pypi
 RUN install-tool poetry 1.4.2
+# renovate: datasource=pypi
+RUN install-tool pdm 2.6.1
```
Comment (viceice) [inline test/Dockerfile.jammy:141]: please also add it to python test dockerfile

File: test/python/Dockerfile [hunk 3]
```diff
@@ -193,6 +193,20 @@ RUN set -ex; \
...
+#--------------------------------------
+# test pdm: pdm (multiple python)
+#--------------------------------------
+FROM build as test-pdm
+
+USER 1000
+
+# make as latest
+# renovate: datasource=github-releases packageName=containerbase/python-prebuild
+RUN install-tool python 3.11.3
```
Comment (viceice) [inline test/python/Dockerfile:205]: isn't python already on build stage?

File: src/usr/local/buildpack/tools/v2/pdm.sh
Comment (viceice) [reply src/usr/local/buildpack/tools/v2/pdm.sh:1]: yes, the python util defines the required functions and uses the passed env values to install the expected tool and version

File: test/python/Dockerfile (hunk 3 above)
Comment (secustor) [reply test/python/Dockerfile:205]: The tests for pipenv do also install python on each stage. https://github.com/containerbase/base/pull/973/files#diff-8722b87dfd4067380b8581bb8d4727b2bedaa8d620571984262fabc897af38b3R183

**Review by rarkins — APPROVED** [review]

File: test/python/Dockerfile (hunk 3 above)
Comment (viceice) [reply test/python/Dockerfile:205]: not all, only some to test older python. so remove to speed up tests

**Review by viceice — APPROVED** [review]

**Pull request merged**